File: Backend/src/main/java/com/inn/cafe/service/BillService.java
package com.inn.cafe.service;

import com.inn.cafe.POJO.Bill;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Map;

public interface BillService {
    ResponseEntity<String> generateReport(Map<String, Object> requestMap);
    ResponseEntity<List<Bill>> getBills();

    ResponseEntity<byte[]> getPdf(Map<String, Object> requestMap);

    ResponseEntity<String> delete(Integer id);

    ResponseEntity<String> updateStatus(Map<String, String> requestMap, Integer id);
    
    ResponseEntity<String> updateProductDetails(Map<String, Object> requestMap);
}

File: Frontend/src/app/services/bill.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root',
})
export class BillService {
  url = environment.apiUrl;
  constructor(private httpClient: HttpClient) {}

  generateReport(data: any) {
    return this.httpClient.post(this.url + '/bill/generateReport', data, {
      headers: new HttpHeaders().set('Content-Type', 'application/json'),
    });
  }

  getPdf(data: any): Observable<Blob> {
    return this.httpClient.post(this.url + '/bill/getPdf', data, {
      responseType: 'blob',
    });
  }

  getBills() {
    return this.httpClient.get(this.url + '/bill/getBills');
  }

  getBillsByStatus(status: string) {
    return this.httpClient.get(this.url + '/bill/getBillsByStatus/' + status);
  }

  updateStatus(bill: any, status: string) {
    return this.httpClient.post(this.url + '/bill/updateStatus/' + bill, { status: status }, {
      headers: new HttpHeaders().set('Content-Type', 'application/json'),
    });
  }

  delete(bill: any) {
    return this.httpClient.post(this.url + '/bill/delete/' + bill, {
      headers: new HttpHeaders().set('Content-Type', 'application/json'),
    });
  }

  updateProductDetails(data: any) {
    return this.httpClient.post(this.url + '/bill/updateProductDetails', data, {
      headers: new HttpHeaders().set('Content-Type', 'application/json'),
    });
  }
}
